IdentityCore/src/validation/MSIDAadAuthorityCache.m

```diff
@@ -74,63 +53,51 @@ + (MSIDAadAuthorityCache *)sharedInstance
     return singleton;
 }
 
-- (BOOL)processMetadata:(NSArray<NSDictionary *> *)metadata
+- (void)processMetadata:(NSArray<NSDictionary *> *)metadata
    openIdConfigEndpoint:(NSURL *)openIdConfigEndpoint
-              authority:(NSURL *)authority
+              authority:(MSIDAADAuthority *)authority
                 context:(id<MSIDRequestContext>)context
-                  error:(NSError **)error
+             completion:(void (^)(BOOL result, NSError *error))completion
 {
-    if (metadata != nil)
-    {
-        CHECK_CLASS_TYPE(metadata, NSArray, @"JSON metadata from authority validation is not an array");
-    }
-    
-    [self getWriteLock];
-    BOOL ret = [self processImpl:metadata authority:authority openIdConfigEndpoint:openIdConfigEndpoint context:context error:error];
-    pthread_rwlock_unlock(&_rwLock);
-    
-    return ret;
+    dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), ^{
+        NSError *error;
+        BOOL result = [self processImpl:metadata authority:authority openIdConfigEndpoint:openIdConfigEndpoint context:context error:&error];
+        dispatch_async(dispatch_get_main_queue(), ^{
+            completion(result, error);
+        });
+    });
 }
 
 static BOOL VerifyHostString(NSString *host, NSString *label, BOOL isAliases, id<MSIDRequestContext> context, NSError * __autoreleasing *error)
 {
     CHECK_CLASS_TYPE(host, NSString, ([NSString stringWithFormat:@"\"%@\" in JSON authority validation metadata must be %@", label, isAliases ? @"an array of strings" : @"a string"]));
     
-    @try
-    {
-        // Run this through urlForPreferredHost to make sure it does not throw any exceptions
-        urlForPreferredHost([NSURL URLWithString:@"https://fakeurl.contoso.com"], host);
-        
-        return YES;
-    }
-    @catch (NSException *ex)
+    // Run this through urlForPreferredHost to make sure it does not return any errors.
+    NSError *err;
+    [[NSURL URLWithString:@"https://fakeurl.contoso.com"] msidURLForPreferredHost:host context:context error:&err];
```

Comment:
why did you remove the try/catch from here?

Reply:
try/catch moved to msidURLForPreferredHost